docs/02_database/05_database-interface/01_entitydb.mdx

```diff
@@ -662,62 +747,417 @@ final var subscription = db.subscribe(Trade.class)
+We do this to reduce the load on high volume systems.
+
+For example, if a trade is inserted, and then modified rapidly in quick succession, oftentimes we are not interested
+in all the intermediate states.
+It will be easier for the system to just press a single update, so the insert and modify can get merged into a single
+insert update.
+
+### Batching phases
+
+Batching happens in two phases:
+1. we gather updates
+2. at an interval we receive a tick and we aggregate and flush updates
+
+So what does this mean in practice?
+The batching period is always set globally, i.e. for an update queue subscription or on a data server query.
+Never on the individual row level.
+
+The goal of this batching is to minimise work, and to publish updates that are accurate, rather than to publish every change.
+
+In this regard it is better to think in terms of a batching interval, rather than a batching period, which looks something like this:
+
+```
+...
+*
+|---- gathering phase ----| (batching period)
+*
+*
+*
+*
+*
+|---- flushing phase -----| (tick happens here)
+* aggregate updates
+* publish updates
+|---- gathering phase ----| (next batching period)
+*
+*
+...
+```
+
+### Batching is non-deterministic
+
+During our gathering phase, we listen for updates; and during the flushing phase, we aggregate and publish.
+
+Let's look at an example:
+
+Imagine we insert a trade with quantity 100. Within milliseconds, we update this trade and set the quantity to 200.
+
+Depending on when the tick happens, you could either get two updates:
+
+```
+|---- gathering phase ----|
+*
+*
+*
+*
+* insert (qty 100)
+|---- flushing phase -----|
+* aggregate updates
+* publish updates insert (qty 100)
+|---- gathering phase ----|
+* modify (qty 200)
+*
+*
+*
+*
+|---- flushing phase -----|
+* aggregate updates
+* publish updates modify (qty 200)
+|---- gathering phase ----| (batching period)
+*
+*
+```
+
+1. Insert trade, qty = 100
+2. Modify trade, qty = 200
+
+Or you could receive a single update:
+
+```
+|---- gathering phase ----|
+*
+*
+*
+* insert (qty 100)
+* modify (qty 200)
+|---- flushing phase -----|
+* aggregate updates
+* publish updates insert (qty 200)
+```
+
+1. Insert trade, qty = 200
+
+Depending on timing, either case is possible, and the smaller the gap between the insert and modify, the more likely the second scenario becomes.
+However, even if these operations happen immediately after each other, we could still receive two updates.
+
+When we get two updates of the same record within the batching interval we will always aggregate the updates.
+This could even mean that updates will be disregarded all together, for example if we receive an insert and a delete within the batching interval.
+
+This will always be non-deterministic, as we don't provide any guarantees.
+
+#### Batching at different levels
+
+The other thing to note is that batching happens on both the data server and the update queue level.
```

Comment:
we talk about the dataserver here, but it is entitydb documentation. Maybe worth talking about the data server as a great use case for update queue batching and explaining it from that point of view. Then we can link to dataserver docs from here as we mention things like LMDB which might not be obvious from just the content here.

Reply:
maybe more of a general link to dataserver documentation